### PR TITLE
Update users endpoint to new API specs

### PR DESCRIPTION
**Pull request by jakecomp**
# C2-45 Update Users Endpoint 

## What's Changed 
In this PR we updated the /users endpoint to reflect the new agreed upon API specs. We removed the delete users endpoint since these are no longer required. We updated the user's json and we also changed permissions logic. Before, a user couldn't even add in their own preferences because only admin users can do POST/PUT requests. We have changed this logic so that users can also make changes to only themselves (i.e like adding a preference)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -131,10 +131,6 @@ func handleUserRequests(router *mux.Router) {
 	}).Methods(http.MethodPost)
 
 	// Users CRUD Operations
-	router.HandleFunc("/users", func(w http.ResponseWriter, r *http.Request) {
```
Comment (AnthonyRodway) [inline main.go]: Is there an updated API doc cause in the API and Data Reference doc it looks like the user creation endpoint is still there?

Comment (jakecomp) [reply main.go]: Whoops... Yep I misread that. Will fix right away!

Comment (jakecomp) [reply main.go]: Okay, we can add users again

**Review by AnthonyRodway — APPROVED** [review]
👍

**Review by jberryman566 — APPROVED** [review]

**Pull request merged**